codex/blockexchange/engine/engine.nim

```diff
@@ -365,18 +398,22 @@ proc blocksDeliveryHandler*(
 
     validatedBlocksDelivery.add(bd)
 
-  await b.resolveBlocks(validatedBlocksDelivery)
+  await self.resolveBlocks(validatedBlocksDelivery)
   codex_block_exchange_blocks_received.inc(validatedBlocksDelivery.len.int64)
 
-  let peerCtx = b.peers.get(peer)
+  let peerCtx = self.peers.get(peer)
 
   if peerCtx != nil:
-    await b.payForBlocks(peerCtx, blocksDelivery)
+    await self.payForBlocks(peerCtx, blocksDelivery)
     ## shouldn't we remove them from the want-list instead of this:
     peerCtx.cleanPresence(blocksDelivery.mapIt(it.address))
```

Comment:
Just above the highlighted lines we are calling `resolveBlocks`. `resolveBlocks` will call `cancelBlocks` which should take care for removing the relevant presence entries.